specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/compute.json

```diff
@@ -9610,10 +9613,30 @@
             "name": "VirtualMachineSizeTypes",
             "modelAsString": true
           }
+        },
+        "vmSizeProperties": {
+          "$ref": "#/definitions/VMSizeProperties",
+          "description": "Specifies the VM Size Properties for customizing VM sizes."
         }
       },
       "description": "Specifies the hardware settings for the virtual machine."
     },
+    "VMSizeProperties": {
+      "type": "object",
+      "properties": {
+        "vCPUsAvailable": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Indicates the number of vCPUs available for the VM. <br><br> When this property is not specified in the request body, default behavior is to set it to the default number of vCPUs available for that VM size."
+        },
+        "vCPUsPerCore": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Indicates the vCPU to physical core ratio. Default value is set to the vCPUsPerCore available for the VM Size. <br><br> Setting this property to 1 also means that hyper-threading is disabled."
```

Comment:
Indicates the vCPU to physical core ratio. <br><br>When this property is not specified in the request body the default behavior is set to the value of vCPUsPerCore for the VM Size  exposed in api response of List all available virtual machine sizes in a region <br><br> Setting this property to 1 also means that hyper-threading is disabled.

Reply:
ack